### PR TITLE
REF: simplify _iterate_slices

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -142,8 +130,8 @@ def pinner(cls):
 class SeriesGroupBy(GroupBy):
     _apply_whitelist = base.series_apply_whitelist
 
-    def _iterate_slices(self) -> Iterable[Tuple[Optional[Hashable], Series]]:
-        yield self._selection_name, self._selected_obj
```
Comment (jorisvandenbossche) [inline pandas/core/groupby/generic.py]: I don't know anything about this code, but: according to the property just below `self._selection_name` is not always equal to `self.obj.name` (which is what you replaced this with below)

Comment (jbrockmendel) [reply pandas/core/groupby/generic.py]: Yah its a bit roundabout, but it turns out that for the SeriesGroupBy they are equivalent

Comment (jorisvandenbossche) [reply pandas/core/groupby/generic.py]: OK!

**Comment by jreback** [comment]
thanks

**Pull request merged**